lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -225,24 +225,25 @@ def discover_resources
         end
       end
       resources
+    rescue InvalidTemplateError => e
+      debug_msg = paragraph_for_invalid_templates(e.message, filenames: [e.filename], content: e.content)
+      @logger.summary.add_paragraph(debug_msg)
```

Comment:
Why log here instead of `split_templates`

Reply:
> Why log here instead of split_templates

I think I did that because one of the three places that can raise this is the `rescue` of `split_templates`, so the alternative is:

```ruby
    def split_templates(filename)
      ...
    rescue Psych::SyntaxError => e
      log_invalid_templates(e.message, filenames: [filename], content: rendered_content)
      raise FatalDeploymentError, "Failed to render and parse template"
    rescue InvalidTemplateError => e
      log_invalid_templates(e.message, filenames: [e.filename], content: e.content)
      raise FatalDeploymentError, "Failed to render and parse template"
    end
```

It's a bit of duplication, but I don't feel strongly about it. Do you think that would be easier to follow?

Reply:
I think moving it to split_templates is easer to read / understand.